sky/resources.py

```diff
@@ -291,6 +291,18 @@ def __repr__(self) -> str:
 
         return f'{cloud_str}({hardware_str})'
 
+    @property
+    def repr_with_region_zone(self) -> str:
+        region_str = ''
+        if self.region is not None:
+            region_str = f', region={self.region}'
+        zone_str = ''
+        if self.zone is not None:
+            zone_str = f', zone={self.zone}'
+        repr_str = repr(self)
+        repr_str = repr_str.replace(')', f'{region_str}{zone_str})')
```

Comment:
nit: Would it be more robust to assert that `repr[-1] == ')'` and then make `repr_str = repr_str[:-1] + f'{region_str}{zone_str})'`? Just to avoid replacing any other `)` that may be present in the repr.